### PR TITLE
Dropping construction material cancels build action bar

### PR DESCRIPTION
**Pull request by pgmzeta**
## About the PR 
Checks to make sure you're holding your building material throughout the entire build action-bar

## Why's this needed? 
Currently, if the sheets you have in-hand are dropped/thrown/disarmed you continue building like nothing happened.

### CODE REVIEW

**Comment by pgmzeta** [comment]
tested with humans/cyborgs/ghostdrones. not sure what else is using these.

File: code/datums/controllers/action_controls.dm [hunk 1]
```diff
@@ -661,12 +661,22 @@ var/datum/action_controller/actions
 		. = ..()
 		if(QDELETED(sheet) || sheet.amount < cost)
 			interrupt(INTERRUPT_ALWAYS)
+		if (ismob(owner))
+			var/mob/M = owner
+			if(!equipped_or_holding(sheet, M))
+				interrupt(INTERRUPT_ALWAYS)
```
Comment (BatElite) [inline code/datums/controllers/action_controls.dm:667]: Couldn't you just do `equipped_or_holding(sheet, owner)` directly?

Comment (pgmzeta) [reply code/datums/controllers/action_controls.dm:667]: was getting errors with that approach as `owner` is typed as an atom, so casting here is needed as `equipped` is a proc on `mob`

Comment (TobleroneSwordfish) [reply code/datums/controllers/action_controls.dm:667]: that's not how casting works, the ismob check is sufficient, the cast does nothing

Comment (pgmzeta) [reply code/datums/controllers/action_controls.dm:667]: when i don't cast it to a mob i get an error on build:
```code\datums\controllers\action_controls.dm:665:error: owner.equipped: undefined proc```

Comment (pgmzeta) [reply code/datums/controllers/action_controls.dm:666]: did this in [eaa19ba](https://github.com/goonstation/goonstation/pull/10692/commits/eaa19bac6d5d80731f0d36a6826da3810c9382c9)

Comment (pgmzeta) [reply code/datums/controllers/action_controls.dm:666]: beepsky mad :( https://github.com/goonstation/goonstation/actions/runs/3074569633/jobs/4967432880
```code/datums/controllers/action_controls.dm, line 665, column 8:
error: undefined proc: "equipped" on /atom```

where am I going wrong here?

Comment (Tarmunora) [reply code/datums/controllers/action_controls.dm:667]: the cast to mob is required because we don't use `:` everywhere

**Pull request merged**